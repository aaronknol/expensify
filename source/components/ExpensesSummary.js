import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpenseTotal = numeral(props.expensesTotal/100).format('$0,0.00');

    return (
        <p>Viewing {props.expenseCount} {expenseWord} totalling {formattedExpenseTotal} </p>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
}
const ConnectedExpenseSummary = connect(mapStateToProps)(ExpensesSummary);

export default ConnectedExpenseSummary;