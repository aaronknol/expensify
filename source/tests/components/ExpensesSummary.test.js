import React from 'React';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary.js'
import expenses from '../fixtures/expenses';
import expensesTotal from '../../selectors/expenses-total';

test('should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length} expensesTotal={expensesTotal(expenses)} />);
    expect(wrapper.find('p').text()).toBe('Viewing 3 expenses totalling $1,141.95 ');
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with single total', () => {
    var testExpenses = [expenses[0]];
    console.log('1. ', testExpenses.length);
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={expensesTotal([expenses[0]])} />);
    expect(wrapper.find('p').text()).toMatch('Viewing 1 expense totalling $1.95');
    expect(wrapper).toMatchSnapshot();
});
