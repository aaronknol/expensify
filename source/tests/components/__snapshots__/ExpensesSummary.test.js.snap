// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<p>
  Viewing 
  3
   
  expenses
   totalling 
  $1,141.95
   
</p>
`;

exports[`should render ExpenseSummary with single total 1`] = `
<p>
  Viewing 
  1
   
  expense
   totalling 
  $1.95
   
</p>
`;
